### DISCUSSION TITLE
词表

### DISCUSSION THREAD

**Discussion by xv44586** [OpenBMB/CPM-Live#21] [Q&A 问答] [answered]
1.词表内都词是如何确定的？包括词表大小，词的来源，挑选策略等
2.CPMLivetokenizer 中，首先使用了jieba 进行分词，然后再进行tokenzier，我将词表与jieba 词典进行了对比，词表内有一千个词不在jieba 的词典内，这些词只有一些场景下由HMM “切” 出来，这样这些词的token 将会非常的低频，是不是有点问题？

**Comment by jayzzhou-thu** [comment, marked as answer]
您好，我们是采用这样的方式得到的词表：

1. 我们使用jieba分词来确定词的边界，通过jieba分出来的词打成单字重新做bpe。
2. bpe出来的结果不一定是纯词，有可能是字、字词或者词，然后词表中的词是bpe个数设置到30000附近的结果。

希望能够回答您的问题。此外如果您有改进意见或建议，也可以通过“模型倡议”的方式提出，我们可以考虑后续引入到模型训练中～
